### ISSUE TITLE
Api Gateway : missing required argument

### ISSUE THREAD

**Issue by asrivastava2017** [hashicorp/terraform#18644]
Hi Team,

I was using the variable block for the deployment of api gateway : 
resource "aws_api_gateway_deployment" "myapideployment" {
  depends_on = [
    "aws_api_gateway_integration.res1integration",
    "aws_api_gateway_integration.res2integration",
  ]

  rest_api_id = "${aws_api_gateway_rest_api.myapi.id}"
  stage_name  = "deploytest"

  variables {
    version = "${var.version}"
  }

  lifecycle {
    create_before_destroy = true
  }
}

I am calling this as a module.
Till sometime back it was working fine but now it started to give below error : 

[0m[1mInitializing the backend...[0m
[0m[32m
Successfully configured the backend "s3"! Terraform will automatically
use this backend unless the backend configuration changes.[0m
[31m
[1m[31mError: [0m[0m[1mmodule "myapigateway": missing required argument "version"[0m

any idea ??

**Comment by asrivastava2017** [comment]
the value of the version is the combined commit hash of api gateway main.tf and variable.tf

**Comment by apparentlymart** [comment]
Hi again @asrivastava2017!

I didn't hear back from you, so I'm going to assume that the problem was the fact that `version` became a reserved variable name as I expected.

I just checked with v0.12.0-alpha2 that indeed there is now a proper error message for this situation to make the cause more obvious:

```
Error: Invalid variable name

  on module-version-reserved.tf line 1, in variable "version":
   1: variable "version" {

The variable name "version" is reserved due to its special meaning inside
module blocks.
```

This change is merged in master and will be included in the forthcoming v0.12.0 final release, so I'm going to close this out now. Thanks for reporting this, and sorry the cause wasn't clearer in Terraform's output.
